### PR TITLE
Uniprotkb entry - initial

### PR DESCRIPTION
**Pull request by xwatkins**
This represents the first step in displaying data in the entry page. From then on it will be a matter of adding new viewers under the model. In this PR:
- Adding a new `model` directory to handle everything related with display of the data both in results and entry pages
- Adding a route for the uniprotkb entry page
- Add a React hook based component for data fetching

### CODE REVIEW

File: src/model/FieldToViewMappings.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  [index: string]: (row: any) => JSX.Element | undefined;
+} = {
+  accession: (row: { primaryAccession: string }) => (
+    <SimpleView
+      termValue={row.primaryAccession}
+      linkTo={`/uniprotkb/${row.primaryAccession}`}
+    />
+  ),
+  id: (row: { uniProtId: string }) => <SimpleView termValue={row.uniProtId} />,
+  protein_name: (data: ProteinNamesData) => {
```
Comment (dlrice) [inline src/model/FieldToViewMappings.tsx]: We are using `row` and `data` interchangeably in this function, could be easier to read if we pick only one.

File: src/model/ProteinNames.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      };
+    };
+    submissionNames?: [
+      {
+        fullName: {
+          value: string;
+        };
+      }
+    ];
+    shortNames?: [
```
Comment (dlrice) [inline src/model/ProteinNames.tsx]: In `modelData.json`, `shortNames` and `ecNumbers` are properties of `proteinDescription.recommendedName` but the paths in `ProteinNames.tsx` assume they are properties of `proteinDescription` which means they're not being accessed and rendered.

File: src/App.tsx [hunk 3]
```diff
@@ -2,16 +2,41 @@ import React from 'react';
...
 
 const App = () => (
   <FranklinSite>
     <Router basename={BASE_URL}>
       <Switch>
-        <Route path="/" exact render={HomePage} />
-        <Route path="/uniprotkb" component={ResultsPage} />
+        <Route
+          path="/"
+          exact={true}
```
[the remark this answers is not in the document]
Comment (xwatkins) [reply src/App.tsx]: Our current tslint configuration enforces it https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md

File: src/App.tsx [hunk 4]
```diff
@@ -2,16 +2,41 @@ import React from 'react';
...
+          render={() => (
+            <BaseLayout isHomePage={true}>
```
[the remark this answers is not in the document]
Comment (xwatkins) [reply src/App.tsx]: See above

File: src/entry/Entry.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+interface EntryProps extends RouteComponentProps<MatchParams> {}
+
+const Entry: React.FC<EntryProps> = ({ match }) => {
+  const url = apiUrls.entry(match.params.accession);
+  const data = useDataApi(url);
+  if (Object.keys(data).length <= 0) {
+    return null;
+  }
+  const entryData = data.results[0];
```
[the remark this answers is not in the document]
Comment (xwatkins) [reply src/entry/Entry.tsx]: Added conditional checks

File: src/model/FreeText.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+type FreeTextDataProps = {
+  data: FreeTextData;
+  type: FreeTextType;
+};
+
+export const FreeText: React.FC<FreeTextDataProps> = ({ data, type }) => {
+  const freeTextData = data.comments
+    .filter(d => d.commentType === type)
+    .map((item, i) => (
+      <p key={i}>{item.texts.map(itemText => itemText.value)}</p>
```
[the remark this answers is not in the document]
Comment (xwatkins) [reply src/model/FreeText.tsx]: As these items ares static (ie not computed) and can't be reordered or filtered (like rows etc...) it doesn't really matter in this instance. However, as multiple `FreeText` components will be used you're right that it's important to distinguish them: I've added `freetext_${i}_${type}`

File: src/model/FieldToViewMappings.tsx (hunk 1 above)
Comment (xwatkins) [reply src/model/FieldToViewMappings.tsx]: Good point, `data` it is!

File: src/model/FreeText.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (dlrice) [reply src/model/FreeText.tsx]: Sounds good. Will there only be one type shown on a page at once?

[the remark this answers is not in the document]
Comment (xwatkins) [reply src/model/FreeText.tsx]: Yes, only one of each type per entry

**Comment by xwatkins** [comment]
Model updated, should be ready for re-review

**Pull request merged**